docs/adr/ADR-003-async-only-api-with-blocking-get.md

```diff
@@ -0,0 +1,449 @@
+Content-Type: application/json
+
+{
+  "workspaces": [...],
+  "dry_run": true
+}
+
+Response: 202 Accepted
+{
+  "task_id": "550e8400-e29b-41d4-a716-446655440000",
+  "status": "queued",
+  "status_url": "/api/v1/integrations/slack-usergroups/reconcile/550e8400-e29b-41d4-a716-446655440000"
+}
+```
+
+**Characteristics:**
+
+- Always returns `202 Accepted`
+- Always queues background task
+- Returns immediately (no waiting)
+- Works for both dry-run and apply modes
+
+### GET: Retrieve Result (Blocking or Non-Blocking)
+
+GET request retrieves task status and result. Behavior depends on `timeout` parameter.
+
+#### Non-Blocking (Default - No Timeout)
+
+```http
+GET /api/v1/integrations/slack-usergroups/reconcile/550e8400-e29b-41d4-a716-446655440000
+Authorization: Bearer <token>
+
+Response: 200 OK (always, returns current status)
+{
+  "status": "pending",  // or "success", "failed"
+  "actions": [],
+  "applied_count": 0,
+  "errors": null
+}
+```
+
+#### Blocking (With Timeout)
+
+```http
+GET /api/v1/integrations/slack-usergroups/reconcile/550e8400-e29b-41d4-a716-446655440000?timeout=60
+Authorization: Bearer <token>
+
+Response: 200 OK (when complete within timeout)
+{
+  "status": "success",  // or "failed"
+  "actions": [
+    {"action_type": "update_users", "workspace": "...", ...},
+    {"action_type": "update_channels", ...}
+  ],
+  "applied_count": 5,
+  "errors": null
+}
+
+Response: 408 Request Timeout (if still pending after timeout)
+{
+  "detail": "Task still pending after 60s"
+}
+```
+
+**Characteristics:**
+
+- **Without timeout (default)**: Returns immediately with current status (pending/success/failed)
+- **With timeout**: Blocks until task completes OR timeout expires
+- `timeout` query parameter: Optional (default: None = non-blocking, max: from config)
+- Returns `200 OK` with status+result when complete
+- Returns `408 Request Timeout` only in blocking mode if still pending
+- Polls background task status internally (blocking mode only)
+
+## Typical Usage Patterns
+
+### MR Validation
+
+```python
+# reconcile/slack_usergroups_api.py (called from GitLab MR job)
+
+# 1. Queue task
+response = api_client.post("/reconcile", json={
+    "workspaces": desired_state,
+    "dry_run": True,
+})
+
+task_id = response["task_id"]
+status_url = response["status_url"]
+
+# 2. Block until result (with timeout)
+result = api_client.get(status_url, params={"timeout": 120})
+
+# 3. Display actions in MR comment
+for action in result["actions"]:
+    print(f"Would {action['action_type']}: {action['usergroup']}")
+```
+
+### Production Run (Fire-and-Forget)
+
+```python
+# reconcile/slack_usergroups_api.py (called from scheduled job)
+
+# Queue task
+response = api_client.post("/reconcile", json={
+    "workspaces": desired_state,
+    "dry_run": False,
+})
+
+task_id = response["task_id"]
+logging.info(f"Queued reconciliation: {task_id}")
+
+# Don't wait for result - let it run in background
```

Comment:
Should we define patterns here how to react if API is unhealthy? Thinking about queuing more tasks than getting done in the same time, i.e., api overload.

Reply:
This must be done in another ADR in the future.